src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -364,6 +368,15 @@ private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int
                 startLength = 2;
             }
 
+            if (token.IsKind(SyntaxKind.SingleLineRawStringLiteralToken, SyntaxKind.MultiLineRawStringLiteralToken))
+            {
+                var tokenStart = token.ToString().TakeWhile(s => s == '"');
+                startLength = tokenStart.Count();
+
+                return position == token.Span.End &&
+                    (token.Span.Length == startLength || (token.Span.Length > startLength && !token.ToString().EndsWith(string.Concat(tokenStart))));
+            }
```

Comment:
hrmmmmmmm  something seems fishy here.  we likely want to talk about what's going on here and if it's necessary.  i'm guessing we can answer this question without having to stringigy the token.  note: the existing code seems uber suspect to me as well :)